label_studio/projects/functions/next_task.py

```diff
@@ -141,13 +143,36 @@ def get_not_solved_tasks_qs(user, project, prepared_tasks, assigned_flag, queue_
         user_postponed_tasks = postponed_drafts.distinct().values_list('task__pk', flat=True)
         not_solved_tasks = not_solved_tasks.exclude(pk__in=user_postponed_tasks)
 
-    # if annotator is assigned for tasks, he must to solve it regardless of is_labeled=True
+    prioritized_on_agreement = False
+    # if annotator is assigned for tasks, he must solve it regardless of is_labeled=True
     if not assigned_flag:
-        not_solved_tasks = not_solved_tasks.filter(is_labeled=False)
+        # include tasks that have been completed if their agreement is not at threshold if threshold setting is set
+        lse_project = getattr(project, 'lse_project', None)
+        if (
+            lse_project
+            and flag_set('fflag_feat_optic_161_project_settings_for_low_agreement_threshold_score_short')
+            and lse_project.agreement_threshold is not None
+            and get_tasks_agreement_queryset
+            and user.is_annotator
+        ):
+            not_solved_tasks = (
+                get_tasks_agreement_queryset(not_solved_tasks)
+                # include tasks that are not labeled or are labeled but fall below the agreement threshold
+                .filter(
+                    Q(_agreement__lt=lse_project.agreement_threshold, is_labeled=True) | Q(is_labeled=False)
```

Comment:
Not a question for right now, but this line struck me as a bit odd. When would we be considering  any task as labeled that is also under the agreement_threshold? Is that even possible? If we didn't require that check and could use the agreement_threshold ubiquitously that would surely streamline this call further.

Reply:
There's a configurable max annotations. If it reaches that max number of annotations even through it falls below the threshold, it's considered annotated